peps/pep-9999.rst

```diff
@@ -0,0 +1,233 @@
+PEP: 9999
+Title: Syntactic sugar for named arguments
+Author: Joshua Bambrick <jbambrick@google.com>, Chris Angelico <rosuav@gmail.com>
+Sponsor: Guido van Rossum <guido@python.org>
+Discussions-To: https://discuss.python.org/t/syntactic-sugar-to-encourage-use-of-named-arguments/36217
+Status: Draft 
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 28-Oct-2023
+Post-History: 14-Oct-2023
+
+Abstract
+==================
+
+This PEP proposes introducing syntatic sugar ``f(x=)`` for the common
+pattern where a named argument is the same as the name of the variable
+corresponding to its value ``f(x=x)``.
+
+Motivation
+==================
+Keyword argument syntax can become needlessly repetitive and verbose.
+
+Consider the following call:
+::
+
+    my_function(
+      my_first_variable=my_first_variable,
+      my_second_variable=my_second_variable,
+      my_third_variable=my_third_variable,
+    )
+
+The case of a keyword argument name matching the variable name of its value is prevalent among all major Python libraries. This verbosity and redundancy discourages use of named arguments and reduces readability by increasing visual noise.
+
+Rationale
+==================
+There are two ways to invoke a function with arguments: by position and by keyword. Keyword arguments confer many benefits by being explicit, thus increasing readability and minimising the risk of inadvertent transposition. On the flipside, positional arguments are often used simply to minimise verbosity and visual noise.
+
+We contend that a simple syntactic sugar used to simplify this common pattern which would confer numerous benefits:
+
+Encourages use of named variables
+------------------------------------
+This syntax would encourage the use of named variables, thereby increasing readability (*explicit is better than implicit*) and reducing bugs from argument transposition.
+
+Reduces verbosity
+------------------------------------
+By minimising visual noise and in some cases lines of code, we can increase readability (*readability counts*).
+
+Encourages consistent variable names
+------------------------------------
+A common problem is that semantically identical variables have different names depending on their contexts. This syntax would encourage authors to use the same variable name when calling a function as the argument name, which would increase consistency of variable names used and hence also *readability*.
+
+Naturally extends to dictionaries
```

Comment:
The heading "Naturally extends" will imply that this syntax could be used in a dictionary display, but what you're proposing is not that. Not sure how to word it, but what you're giving is an elegant use of the `dict()` constructor to achieve this goal without extending anything syntactically.

Reply:
I've renamed this to 'Applicability to dictionary construction'